src/App.js

```diff
@@ -109,230 +121,240 @@ class Root extends React.Component {
           // remove all invalid signatures
           // arg.data is a list of uids with invalid signatures
           arg.data.forEach(uid => {
-            tokens.removeTokenSignature(uid.toString('hex'));
+            tokensUtils.removeTokenSignature(uid.toString('hex'));
           });
         }
       });
     }
-  }
-
-  componentWillUnmount() {
-    if (IPC_RENDERER) {
-      IPC_RENDERER.removeAllListeners('ledger:closed');
-      IPC_RENDERER.removeAllListeners('ledger:manyTokenSignatureValid');
-    }
-  }
 
-  render() {
-    if (this.props.walletStartState === WALLET_STATUS.FAILED) {
-      return <LoadWalletFailed />;
+    // Removes the ledger event listeners on unmount, if necessary
+    return () => {
+      if (IPC_RENDERER) {
+        IPC_RENDERER.removeAllListeners('ledger:closed');
+        IPC_RENDERER.removeAllListeners('ledger:manyTokenSignatureValid');
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    // Fetch the version information before rendering any screens, if possible
+    if (isVersionAllowed === undefined && wallet) {
+      helpersUtils.loadStorageState();
+
+      // We already handle all js errors in general and open an error modal to the user
+      // so there is no need to catch the promise error below
+      versionUtils.checkApiVersion(wallet);
     }
+  }, [isVersionAllowed, wallet]);
```

Comment:
`checkApiVersion` is an async method with a state change, I think we should write this `useEffect` more like whats described [here](https://devtrium.com/posts/async-functions-useeffect) what do you think?

Reply:
As we discussed, this is an important change and will be made on a dedicaded PR. I'll open an issue for that.

Reply:
This is being tracked on issue
- #526